src/sentry/api/serializers/models/release.py

```diff
@@ -62,17 +62,19 @@ def get_users_for_authors(organization_id, authors, user=None):
     }
     """
     # Filter users based on the emails provided in the commits
-    user_emails = list(
-        UserEmail.objects.filter(in_iexact("email", [a.email for a in authors])).order_by("id")
+    user_emails = sorted(
+        UserEmail.objects.get_many_from_cache([a.email.lower() for a in authors], key="email"),
+        key=lambda x: x.id,
     )
-
     # Filter users belonging to the organization associated with
     # the release
-    users = User.objects.filter(
-        id__in={ue.user_id for ue in user_emails},
-        is_active=True,
-        sentry_orgmember_set__organization_id=organization_id,
-    )
+    users = [
+        u
+        for u in User.objects.get_many_from_cache([ue.user_id for ue in user_emails])
+        if u.is_active
+        and OrganizationMember.objects.filter(user=u, organization_id=organization_id).exists()
+    ]
+
```

Comment:
I want to figure out how to cache this better.

Reply:
Yeah we def need to make this work better, right now it's making N queries to check org members. 

It might be best to just cache the results of the query entirely, rather than splitting it into two queries, and have the cache key be something like 
`users:<org_id>:<sha of concatenated user_ids>`